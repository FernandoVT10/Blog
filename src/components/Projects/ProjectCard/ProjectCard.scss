.project {
    cursor: pointer;
    position: relative;
    width: 100%;
    height: 250px;

    &::after {
        content: 'View Project';
        position: absolute;
        display: flex;
        justify-content: center;
        align-items: center;
        text-transform: uppercase;
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        color: white;
        font-weight: bold;
        background: rgba($color: #000000, $alpha: .6);
        opacity: 0;
        transition: .3s;
    }

    &:hover::after {
        opacity: 1;
    }

    &__image {
        position: absolute;
        opacity: 0;
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        object-fit: cover;
        transition: 1s;

        &--active {
            opacity: 1;
        }
    }
}