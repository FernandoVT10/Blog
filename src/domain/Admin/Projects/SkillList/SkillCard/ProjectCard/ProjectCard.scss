.admin-skill-list-project-card {
    display: flex;
    justify-content: space-between;
    padding: 20px;

    &__title {
        color: var(--white);
        font-weight: bold;
        transition: .3s;

        &:hover {
            text-decoration: none;
            color: var(--secondary);
        }
    }
}