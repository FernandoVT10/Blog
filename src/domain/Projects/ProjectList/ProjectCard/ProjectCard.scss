.projects-project-card {
    cursor: pointer;
    position: relative;
    width: 100%;
    height: 250px;
    border: solid 2px var(--secondary);

    &::after {
        content: 'View Project';
        font-size: 1.2em;
        position: absolute;
        display: flex;
        justify-content: center;
        align-items: center;
        text-transform: uppercase;
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        color: var(--black);
        font-weight: bold;
        background: var(--secondary);
        opacity: 0;
        transition: .3s;
    }

    &:hover::after {
        opacity: 1;
    }

    &__image {
        position: absolute;
        opacity: 0;
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        object-fit: cover;
        transition: 1s;

        &--active {
            opacity: 1;
        }
    }
}