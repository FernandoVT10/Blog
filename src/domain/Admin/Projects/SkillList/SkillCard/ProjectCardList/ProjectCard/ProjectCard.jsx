import ConfirmModal from "@/components/ConfirmModal";

import { useState } from "react";
import Link from "next/link";
import { useRouter } from "next/router";

import "./ProjectCard.scss";

const ProjectCard =  ({ project, deleteProject }) => {
    const [confirmModalActive, setConfirmModalActive] = useState(false);
    const router = useRouter();

    const onClose = (confirm) => {
        if(confirm) {
            deleteProject(project._id);
        }

        setConfirmModalActive(false);
    }
    
    return (
        <div className="admin-skill-list-project-card">
            <ConfirmModal
            active={confirmModalActive}
            message={`Are you sure to delete "${project.title}"?`}
            prefix={project._id}
            onClose={onClose} />

            <Link href={`/projects?project=${project._id}`}>
                <a className="admin-skill-list-project-card__title">
                    { project.title }
                </a>
            </Link>

            <div className="admin-skill-list-project-card__actions">
                <Link href={router.pathname} as={`/admin/projects/${project._id}/edit`}>
                    <button
                    className="custom-table__button custom-table__button--edit">
                        <i className="fas fa-edit" aria-hidden="true"></i>
                    </button>
                </Link>

                <button
                className="custom-table__button custom-table__button--delete"
                onClick={() => setConfirmModalActive(true)}>
                    <i className="fas fa-times" aria-hidden="true"></i>
                </button>
            </div>
        </div>
    );
}

export default ProjectCard;