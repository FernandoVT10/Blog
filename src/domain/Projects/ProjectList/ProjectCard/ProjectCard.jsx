import { useEffect, useState } from "react";
import { useRouter } from "next/router";

import "./ProjectCard.scss";

const ProjectCard =  ({ project }) => {
    const [imageActive, setImageActive] = useState(0);
    
    const router = useRouter();

    useEffect(() => {
        const interval = setTimeout(() => {
            setImageActive((imageActive + 1) % project.images.length);
        }, 4000);

        return () => {
            clearInterval(interval);
        };
    }, [imageActive]);

    const handleClick = () => {
        const { query } = router;

        Object.assign(query, { project: project._id });

        router.push({
            pathname: router.pathname,
            query
        });
    };

    return (
        <div className="projects-project-card" onClick={handleClick}>
            {project.images.map((image, index) => {
                const projectClass = imageActive === index ? "projects-project-card__image--active" : "";

                return (
                    <img
                    src={`/img/projects/${image}`}
                    className={`projects-project-card__image ${projectClass}`}
                    alt={project.title}
                    key={index} />
                );
            })}
        </div>
    );
};

export default ProjectCard;