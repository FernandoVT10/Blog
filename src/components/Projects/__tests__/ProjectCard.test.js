import ProjectCard from "../ProjectCard/";
import { act } from "react-dom/test-utils";
import { render } from "react-dom";

const PROJECT_MOCK = {
    _id: 123,
    title: "Test title",
    images: ["test-image-1.jpg", "test-image-2.jpg", "test-image-3.jpg"]
};

let container;

jest.mock("next/router", () => ({
    useRouter: jest.fn(() => ({ query: {} })),
}));

describe("<ProjectCard/> component", () => {
    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });
    
    afterEach(() => {
        document.body.removeChild(container);
        container = null;
    });

    it("Check if renders correctly", () => {
        render(<ProjectCard project={PROJECT_MOCK} />, container);

        expect(container).toMatchSnapshot();
    });

    it("It should change the image class", async () => {
        jest.useFakeTimers();

        await act(async () => {
            render(<ProjectCard project={PROJECT_MOCK} />, container);
        });

        const images = container.querySelectorAll("img");

        expect(images[0].classList.contains("project__image--active")).toBeTruthy();

        act(() => {
            jest.runAllTimers(); 
        });

        expect(images[0].classList.contains("project__image--active")).toBeFalsy();
        expect(images[1].classList.contains("project__image--active")).toBeTruthy();
    });
});