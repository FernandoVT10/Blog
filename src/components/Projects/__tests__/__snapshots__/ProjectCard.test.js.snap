// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectCard/> component Check if renders correctly 1`] = `
<div>
  <div
    class="project"
  >
    <img
      alt="Test title"
      class="project__image project__image--active"
      src="/img/projects/test-image-1.jpg"
    />
    <img
      alt="Test title"
      class="project__image "
      src="/img/projects/test-image-2.jpg"
    />
    <img
      alt="Test title"
      class="project__image "
      src="/img/projects/test-image-3.jpg"
    />
  </div>
</div>
`;
